gc/2020/GC-09-22.md

```diff
@@ -0,0 +1,306 @@
+* Ben Smith
+* Ben Titzer
+* Conrad Watt
+* Daniel Wirtz
+* Derek Schuff
+* Dmitry Bezhetskov
+* Emanuel Ziegler
+* Francis McCabe
+* Jakob Kummerow
+* Keith Miller
+* Lars Hansen
+* Luke Wagner
+* Manos Koukoutos
+* Ross Tate
+* Ryan Hunt
+* Sabine
+* Slava Kuzmich
+* Thomas Lively
+* Tobias Tebbi
+* Wouter Van Oortmersson
+* Zalim Bashorov
+* Zhi An Ng
+
+### Discussion: [Incorporate RTTs into the type](https://github.com/WebAssembly/gc/issues/119) (Ross Tate) [20 min presentation, 20 min discussion]
+
+Ross presenting
+
+TODO(Ross Tate): add link to slides
+
+LW: I get the general tension between i31ref and smi -- what if the engine doesn't need to distinguish them?
+
+RT: specs will because JS SMI is conceptually a JS integer, it’s not supposed to  cast to an i31ref
+
+LW: The idea is that the coercion when the i31 flows in can put it into an i31ref...
+
+RT: you’ve picked a winner in Wasm that will always use i31ref, to interact with Wasm it would convert to Wasm mode
+
+LW: No, it would just happen at the toWebAssembly interaction -- it would not be observable at the JS boundary.
+
+RT: formally in terms of abstraction it is compatible, an alternative is an externref, not a subtype of anyref, i don’t need to do any coercion
+
+LW: Do what? A coercion happens in any case, it happens when a JS value flows into Wasm.
+
+RT: we can do all sorts of coercion at boundary, if externref is not subtype of anyref, you don’t need any coercion
+
+LW: Specifically for SMIs, I don't see a conflict where anyone is getting deoptimized. I don't see a tension for this one.
+
+RT: move on to other examples first
+
+[back to slides -- Centralizing Coordination]
+
+AR: coordination for types is not the same as for exceptions, exceptions are generative in source language, nominal aspects map down
+
+RT: Wait, the generative source in source language are not the same as generative source in Wasm...
+
+AR: in that case it's even easier, either it’s a finite set (C++) or translated down from nominal in source language. Completely different from types. If you compile two modules that use structural types, there is no central instance to define the infinite set of types.
+
+RT: I'll get to all this later… plan to have more discussion after that.
+
+[back to slides -- Centralizing Coordination]
+
+FM: Assertion you have about coordination is not generally applicable. In particular in systems like Java application servers, this a central module but the application server has no knowledge of the applications that it's serving. It's not true that this runtime module can know everything about the application -- it's not knowable.
+
+RT: doesn’t have to know everything, if you have app server and exchange strings, all those strings need to be tagged with rtts.
+
+FM: It's true for strings, but not for other types involved...
+
+RT: Should be for other types as well… 
+
+FM: thats why i said tree, it’s not one node that knows all the type, you have a tree of a graph
+
```

Comment:
@fgmccabe am I correct that this point was meant to be about having a tree of class loaders rather than having a tree data type as reflected here?